plugins/module_utils/foreman_helper.py

```diff
@@ -550,6 +562,15 @@ def ensure_scoped_parameters(self, scope, entity, parameters):
                         'parameters', None, current_parameter, state="absent", entity_spec=parameter_entity_spec, params=scope)
         return changed
 
+    def exit_json(self, **kwargs):
+        if self._after is None or 'invalid' not in self._after:
+            if 'diff' not in kwargs:
+                kwargs['diff'] = {'before': _flatten_entity(self._before or {}, self.entity_spec),
+                                  'after': _flatten_entity(self._after or {}, self.entity_spec)}
```

Comment:
I do not think, this `or` can ever be hit (unless something in the `__init__` calls `exit_json`).

Reply:
sure, `ensure_entity` with `state=absent` will make `self._after` `None` and `_flatten_entity` can't cope with that.

Reply:
(you could argue we can fix that in flatten)

Reply:
(fixed flatten_entity)